src/libraries/System.Net.Sockets/tests/FunctionalTests/TcpClientTest.cs

```diff
@@ -493,6 +493,40 @@ public void Connect_Dual_Success()
             }
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void Connect_MappedV4_Success(bool useConnect)
+        {
+            if (!Socket.OSSupportsIPv6 || !Socket.OSSupportsIPv4)
```

Comment:
Is this check needed?  If both of these were false, wouldn't pretty much all of our networking tests fall over? :)

Reply:
I was thinking about it if I should add this.. The test above just tests for Socket.OSSupportsIPv6  while IPv4 is assumed to exit.  When I add this I was primarily thinking about situation when one AF is available but not the other e.g. case where the duality does not exist. Perhaps that day is still far away. I've also seen some test here this was stated as attribute on the test. I generally don't like the `if (..) return` as that creates lame test passes but I copy that fragment from test above.

Reply:
We pretty much assume IPv4 is always supported; if that ever changes we will have work to do.

Reply:
that make sense. I removed the second check.